### PR TITLE
Add Count and Length properties to [PSCustomobject]

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary
Related #3671 

- Add Count and Length properties to [PSCustomobject].
Now following works with singletons and returns 1:
    ([pscustomobject] @{ foo = 'bar' }).Count
    ([pscustomobject] @{ foo = 'bar' }).Length
- Add tests

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [NA] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

File: test/powershell/engine/ETS/Adapter.Tests.ps1 [hunk 1]
```diff
@@ -112,6 +112,27 @@ Describe "Adapter Tests" -tags "CI" {
...
+        $x.Length | Should Be 1
+
+        $null.Count | Should Be 0
+        $null.Length | Should Be 0
+
+        (10).Count | Should Be 1
+        (10).Length | Should Be 1
+
+        ("a").Count | Should Be 1
+        ("a").Length | Should Be 1
```
Comment (lzybkr) [inline test/powershell/engine/ETS/Adapter.Tests.ps1]: The `Length` property exists for strings, so this isn't (shouldn't be) testing singletons.

File: test/powershell/engine/ETS/Adapter.Tests.ps1 [hunk 2]
```diff
@@ -112,6 +112,27 @@ Describe "Adapter Tests" -tags "CI" {
...
+
+        ([psobject] @{ foo = 'bar' }).Count | Should Be 1
+        ([psobject] @{ foo = 'bar' }).Length | Should Be 1
+
+        ([pscustomobject] @{ foo = 'bar' }).Count | Should Be 1
+        ([pscustomobject] @{ foo = 'bar' }).Length | Should Be 1
```
Comment (lzybkr) [inline test/powershell/engine/ETS/Adapter.Tests.ps1]: You should also test when a `pscustomobject` has a `count` or `length` property to make sure we return the correct value.

**Review by lzybkr — APPROVED** [review]
Approved w/ a couple test suggestions.

File: test/powershell/engine/ETS/Adapter.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [reply test/powershell/engine/ETS/Adapter.Tests.ps1]: Fixed.

File: test/powershell/engine/ETS/Adapter.Tests.ps1 (hunk 2 above)
Comment (iSazonov) [reply test/powershell/engine/ETS/Adapter.Tests.ps1]: Fixed.

**Pull request merged**